app/component/route/RouteStopListContainer.js

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Relay from 'react-relay';
+import RouteStop from './route-stop';
+import groupBy from 'lodash/groupBy';
+import cx from 'classnames';
+import geoUtils from '../../util/geo-utils';
+import config from '../../config';
+
+class RouteStopListContainer extends React.Component {
+  static contextTypes = {
+    getStore: React.PropTypes.func.isRequired,
+  };
+
+  constructor(...args) {
+    super(...args);
+    this.getNearestStopDistance = this.getNearestStopDistance.bind(this);
+    this.onRealTimeChange = this.onRealTimeChange.bind(this);
+  }
+
+  componentDidMount() {
+    this.context.getStore('RealTimeInformationStore').addChangeListener(this.onRealTimeChange);
+
+    if (this.refs.nearestStop) {
+      ReactDOM.findDOMNode(this.refs.nearestStop).scrollIntoView(false);
+    }
+  }
+
+  componentWillUnmount() {
+    this.context.getStore('RealTimeInformationStore').removeChangeListener(this.onRealTimeChange);
+  }
+
+  onRealTimeChange() {
+    this.forceUpdate();
+  }
+
+  getNearestStopDistance(stops) {
+    const state = this.context.getStore('PositionStore').getLocationState();
+
+    if (state.hasLocation === true) {
+      return geoUtils.getDistanceToNearestStop(state.lat, state.lon, stops);
+    }
+    return null;
+  }
+
+  getStops() {
+    const nearest = this.getNearestStopDistance(this.props.pattern.stops);
+    const mode = this.props.pattern.route.type.toLowerCase();
+    const { vehicles } = this.context.getStore('RealTimeInformationStore');
+
+    const vehicleStops = groupBy(vehicles, vehicle => `HSL:${vehicle.next_stop}`);
+
+    const stopObjs = [];
+
+    this.props.pattern.stops.forEach((stop) => {
+      const isNearest = (nearest != null ? nearest.stop.gtfsId : void 0) === stop.gtfsId;
+
+      stopObjs.push(
+        <RouteStop
+          key={stop.gtfsId}
+          stop={stop}
+          mode={mode}
+          vehicles={vehicleStops[stop.gtfsId]}
+          distance={
+            (isNearest && nearest.distance < config.nearestStopDistance.maxShownDistance) ?
+            nearest.distance : null}
+          ref={
```

Comment:
I think you could use a [ref callback](https://facebook.github.io/react/docs/more-about-refs.html#the-ref-callback-attribute) here, i.e. something like `ref={(nearest) => nearest == null ? void 0 : nearest.scrollIntoView(false)}` and not needing to do it in the componentDidMount